sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobLength.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.specialized.cryptography;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.function.Function;
+
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.ENCRYPTION_PROTOCOL_V1;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.ENCRYPTION_PROTOCOL_V2;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.ENCRYPTION_PROTOCOL_V2_1;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.NONCE_LENGTH;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.TAG_LENGTH;
+
+/**
+ * This class provides helper methods for adjusting encrypted downloads.
+ * <p>
+ * RESERVED FOR INTERNAL USE.
+ */
+
+final class EncryptedBlobLength {
```

Comment:
It would awesome if we could add unit tests for this new class to have a quick way to make sure our calculation of this adjusted blob range is correct, especially around some of the corner cases we ran into.

I think we should be able to do something similar to the parameterized tests in `EncryptedBlobRangeTests` and make a new `EncryptedBlobLengthTests` file with cases where we parameterize the encryption version, the encrypted length, and the region length and make sure the expected adjusted blob length is what we expect.